controller/filter.go

```diff
@@ -0,0 +1,59 @@
+package controller
+
+import (
+	"strings"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+)
+
+func updateFunc(e event.UpdateEvent) bool {
+	node, ok := e.ObjectNew.(*corev1.Node)
+	if !ok {
+		return false
+	}
+	return timeToUpdate(node)
+}
+
+// somehow there's a ton of create events
+func createFunc(e event.CreateEvent) bool {
+	node, ok := e.Object.(*corev1.Node)
+	if !ok {
+		return false
+	}
+	return timeToUpdate(node)
+}
+
+// return true because vmss might need to be updated?
+func deleteFunc(e event.DeleteEvent) bool {
+	return true
+}
+
+func genericFunc(e event.GenericEvent) bool {
+	return false
+}
+
+func timeToUpdate(node *corev1.Node) bool {
+	label, ok := node.Labels[lastUpdateLabel]
+	if !ok {
+		return true // let things through the first time
+	}
+	var period time.Duration
+	// if lastUpdate formatted incorrectly, do I let stuff through?
+	lastUpdate, err := time.Parse(time.RFC3339, strings.ReplaceAll(label, ".", ":"))
+	if err != nil {
```

Comment:
Should you log something for the user to figure out his mistake, and correct it?

Reply:
different PR though. keep this one focused on package structure.

Reply:
Yeah I probably should log something.